frontend/src/metabase/query_builder/actions/downloading.ts

```diff
@@ -34,8 +37,64 @@ interface DownloadQueryResultsParams {
   params?: URLSearchParams | string;
 }
 
+export type ResourceType = "question" | "dashcard" | "ad-hoc-question";
+export type ResourceAccessedVia =
+  | "internal"
+  | "public-link"
+  | "static-embed"
+  | "interactive-iframe-embed"
+  | "sdk-embed";
+
+export type DownloadedResourceInfo = {
+  resourceType: ResourceType;
+  accessedVia: ResourceAccessedVia;
+};
+
+const getDownloadedResourceType = ({
+  dashboardId,
+  dashcardId,
+  uuid,
+  token,
+  question,
+}: Partial<DownloadQueryResultsOpts>): DownloadedResourceInfo => {
+  const cardId = question?.id();
+
+  const isInIframe = isWithinIframe();
+
+  if (dashcardId != null && token != null) {
+    return { resourceType: "dashcard", accessedVia: "static-embed" };
+  } else if (dashboardId != null && uuid != null) {
```

Comment:
Nit: since we return in the body of each condition we can remove the `else if` and use use separate `if` statements for each condition.